core/src/main/scala/org/http4s/HttpDate.scala

```diff
@@ -121,4 +123,164 @@ object HttpDate {
   /** Like `fromZonedDateTime`, but throws any parse failures */
   def unsafeFromZonedDateTime(dateTime: ZonedDateTime): HttpDate =
     unsafeFromInstant(dateTime.toInstant)
+
+  /** `HTTP-date = IMF-fixdate / obs-date` */
+  private[http4s] val parser: Parser1[HttpDate] = {
+    import Parser.{char, failWith, pure, string1}
+    import Rfc5234.{digit, sp}
+
+    def mkHttpDate(
+        year: Int,
+        month: Int,
+        day: Int,
+        hour: Int,
+        min: Int,
+        sec: Int): Parser[HttpDate] =
+      try {
+        val dt = ZonedDateTime.of(year, month, day, hour, min, sec, 0, ZoneOffset.UTC)
```

Comment:
Note that the day of week is parsed, but never validated.  Never has been.  Probably never should be.